pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -208,22 +214,34 @@ func GenerateTaskSpec(
 
 		generatedTaskSpec.Steps = append(generatedTaskSpec.Steps, step)
 
-		// Get volumeMounts added to Task's spec.Volumes
-		for _, volumeInBuildStrategy := range containerValue.VolumeMounts {
-			newVolume := true
-			for _, volumeInTask := range generatedTaskSpec.Volumes {
-				if volumeInTask.Name == volumeInBuildStrategy.Name {
-					newVolume = false
+		for _, vm := range containerValue.VolumeMounts {
+			// here we should check that volume actually exists for this mount
+			// and in case it does not, exit early with an error
+			found := false
+			for _, volume := range buildVolumes {
+				if vm.Name == volume.Name {
+					found = true
 				}
 			}
-			if newVolume {
-				generatedTaskSpec.Volumes = append(generatedTaskSpec.Volumes, corev1.Volume{
-					Name: volumeInBuildStrategy.Name,
-				})
+			if !found {
+				return nil, fmt.Errorf("Volume for the Volume Mount %q is not found", vm.Name)
 			}
+
+			existingVolumeMounts = append(
+				existingVolumeMounts,
+				volumes.NewStepVolumeMount(vm.Name, vm.ReadOnly),
+			)
 		}
 	}
 
+	// Add volumes from the strategy to generated task spec
+	// will have to add buildrun volumes here for merging later, too
```

Comment:
Is that comment still valid? I see BuildRun volumes passed in already?

Reply:
Yeah, I forgot to remove it, it was there in the first iteration.